astropy/units/quantity_helper/function_helpers.py

```diff
@@ -1054,11 +1080,11 @@ def _interpret_tol(tol, unit):
 
 
 @function_helper(module=np.linalg)
-def matrix_rank(M, tol=None, *args, **kwargs):
```

Comment:
Note to reviewers: this one is also pretty unique.
The first argument was always allowed to be positional OR keyword, but its named was changed regardless sometimes before numpy 1.22 . Since all currently supported versions call it `A` I changed it to follow numpy, but this is *again* a case where it's impossible to be both correct (blindly follow numpy) *and* 100% backward compatible. I think it's very unlikely that anyone is using this function with a `Quantity` input and passing the first argument as keyword in the wild, so I went with correctness-over-backward compat, but let me know if that's controversial.